module/x/gravity/keeper/ethereum_event_vote.go

```diff
@@ -50,7 +50,11 @@ func (k Keeper) recordEventVote(
 	// Add the validator's vote to this EthereumEventVoteRecord
 	eventVoteRecord.Votes = append(eventVoteRecord.Votes, val.String())
 
-	k.setEthereumEventVoteRecord(ctx, event.GetEventNonce(), event.Hash(), eventVoteRecord)
+	// ignore "old" event votes but still record the nonce for the submitting validator
+	if event.GetEventNonce() > k.GetLastObservedEventNonce(ctx) {
```

Comment:
We should look into the orchestrator/relayer to see if there are improvements that can be made given this change in handling attestations and signatures.

Reply:
The event watch currently only checks its validator's latest event nonce and not the global last observed one. It could just filter out "old" ones if it could query that. Would need that query endpoint you mentioned elsewhere.